frb_rust/src/ffi/web.rs

```diff
@@ -0,0 +1,388 @@
+    // this struct should be deprecated.
+    impl<T> WasmDescribe for Pointer<T> {
+        fn describe() {
+            JsValue::describe()
+        }
+    }
+
+    impl<T> FromWasmAbi for Pointer<T> {
+        type Abi = <JsValue as FromWasmAbi>::Abi;
+
+        unsafe fn from_abi(js: Self::Abi) -> Self {
+            let js = JsValue::from_abi(js);
+            Self::from_js(&js).expect("Attempted to retrieve a nullptr")
+        }
+    }
+
+    impl<T> OptionFromWasmAbi for Pointer<T> {
+        fn is_none(abi: &Self::Abi) -> bool {
+            *abi == 0
+        }
+    }
+
+    impl<T> IntoWasmAbi for Pointer<T> {
+        type Abi = u32;
+
+        fn into_abi(self) -> Self::Abi {
+            self.0.as_ptr() as _
+        }
+    }
+
+    impl<T> OptionIntoWasmAbi for Pointer<T> {
+        fn none() -> Self::Abi {
+            0
+        }
+    }
+}
+pub use pointer::Pointer;
+
+impl IntoDart for () {
+    #[inline]
+    fn into_dart(self) -> DartAbi {
+        JsValue::undefined()
+    }
+}
+macro_rules! delegate {
+    ($( $ty:ty )*) => {$(
+        impl IntoDart for $ty {
+            #[inline]
+            fn into_dart(self) -> DartAbi {
+                DartAbi::from(self)
+            }
+        }
+    )*};
+}
+macro_rules! delegate_buffer {
+    ($( $ty:ty => $buffer:ty )*) => {$(
+        impl IntoDart for $ty {
+            #[inline]
+            fn into_dart(self) -> DartAbi {
+                <$buffer>::from(self.as_slice()).into()
+            }
+        }
+    )*};
+}
+// Orphan rules disallow blanket implementations, so we have to manually delegate here.
+delegate! {
+    bool
+    i8 u8 i16 u16 i32 u32 i64 u64 i128 u128 isize usize
+    f32 f64
+    &str String JsValue
+}
+delegate_buffer! {
+    Vec<i8> => js_sys::Int8Array
+    Vec<u8> => js_sys::Uint8Array
+    Vec<i16> => js_sys::Int16Array
+    Vec<u16> => js_sys::Uint16Array
+    Vec<i32> => js_sys::Int32Array
+    Vec<u32> => js_sys::Uint32Array
+    Vec<f32> => js_sys::Float32Array
+    Vec<f64> => js_sys::Float64Array
+    ZeroCopyBuffer<Vec<i8>> => js_sys::Int8Array
+    ZeroCopyBuffer<Vec<u8>> => js_sys::Uint8Array
+    ZeroCopyBuffer<Vec<i16>> => js_sys::Int16Array
+    ZeroCopyBuffer<Vec<u16>> => js_sys::Uint16Array
+    ZeroCopyBuffer<Vec<i32>> => js_sys::Int32Array
+    ZeroCopyBuffer<Vec<u32>> => js_sys::Uint32Array
+    ZeroCopyBuffer<Vec<f32>> => js_sys::Float32Array
+    ZeroCopyBuffer<Vec<f64>> => js_sys::Float64Array
+}
+
+impl<T: IntoDartExceptPrimitive> IntoDart for Vec<T> {
+    #[inline]
+    fn into_dart(self) -> DartAbi {
+        Array::from_iter(self.into_iter().map(IntoDart::into_dart)).into()
+    }
+}
+
+impl<T: IntoDart> IntoDart for Option<T> {
+    #[inline]
+    fn into_dart(self) -> DartAbi {
+        self.map(T::into_dart).unwrap_or_else(JsValue::null)
+    }
+}
+impl<T> IntoDart for *const T {
+    #[inline]
+    fn into_dart(self) -> DartAbi {
+        (self as u32).into_dart()
+    }
+}
+impl<T> IntoDart for *mut T {
```

Comment:
nit: Is it possible we utilize this intodart, and do not write down into_dart for the Pointer type?